container_pipeline/workers/scan.py

```diff
@@ -71,15 +71,17 @@ def handle_job(self, job):
 
         # if weekly scan, push the job for notification
         if self.job.get("weekly"):
-            scanners_data["action"] = "notify_user"
-            self.queue.put(json.dumps(scanners_data), 'master_tube')
-            self.init_next_phase_data('delivery')
-            self.logger.debug(
-                str.format(
-                    "Weekly scan for {project} is complete.",
-                    project=self.job.get("namespace")
+            # send email of weekly scan only if scanners execution status=true
+            if status:
+                scanners_data["action"] = "notify_user"
+                self.queue.put(json.dumps(scanners_data), 'master_tube')
+                self.init_next_phase_data('delivery')
+                self.logger.debug(
+                    str.format(
+                        "Weekly scan for {project} is complete.",
+                        project=self.job.get("namespace")
+                    )
                 )
```

Comment:
so we are not sending any notification if it weekly scan and scanning failed, right?

Reply:
Right.

For normal build scan failures:

Since the `scan` stage in normal build scenario should not affect final delivery of the image, we move on and deliver the image, that is next [else](https://github.com/CentOS/container-pipeline-service/pull/590/files#diff-92601f9830cf93227125d7e969bc425aR85) statement. The scanners execution in most of th cases will fail only if the image is not present in registry, if the image is built, it will be scan.
In case, where the image is built, but scan stage is failed, the image will be delivered and the email will have
image build details and no scan details.